solvebio/cli/data.py

```diff
@@ -505,6 +506,113 @@ def _download(full_path, local_folder_path, dry_run=False):
             file_.full_path, local_folder_path, file_.filename))
 
 
+def sync(args):
+    """
+    Given a folder or file, download all the files contained
+    within it (not recursive)
+    """
+    return _sync(args.full_path, args.local_path, dry_run=args.dry_run)
+
+
+def _get_remote_directory_recursive(full_path, local_folder_path, remote_files=[]):
+    """Queries a SolveBio Vault path and recursively finds all files.
+
+    full_path (str): SolveBio path where data will be downloaded from.
+    local_folder_path (str): Local folder path where data will be downloaded to.
+    remote_files (list): List used to store queried files in recursive search.
+    """
+    objects = Object.all(glob=full_path, limit=1000)
```

Comment:
One thing to check is whether this returns everything recursively or just at a single "depth".

It might be worth taking a look at some of the examples in here to see if they would be of any use: https://docs.solvebio.com/vaults/basics/#searching-within-vaults

Reply:
Yeah, so my thinking was that the `sync` method while still using the `glob` search would only allow you to do "flat" searches. This can be accomplished by checking if there's a `**` in the query: https://github.com/solvebio/solvebio-python/pull/418/files#diff-d2e9c9fe10a5684bd8ff375bf4aee2594d5cfef7e9da5980759badcc0ded2b40R562

Reply:
Look into using GlobalSearch instead of glob/path/query params.